zerver/forms.py

```diff
@@ -78,6 +87,25 @@ def clean_email(self):
             return data
         raise ValidationError(mark_safe(SIGNUP_STRING))
 
+class RealmCreationForm(forms.Form):
+    # This form determines whether users can
+    # create a new realm. Be careful when modifying the
+    # validators.
+    email = forms.EmailField(validators=[is_inactive,])
+
+    def __init__(self, *args, **kwargs):
+        self.domain = kwargs.get("domain")
+        if "domain" in kwargs:
+            del kwargs["domain"]
+        super(RealmCreationForm, self).__init__(*args, **kwargs)
+
+    def clean_email(self):
+        data = self.cleaned_data['email']
+        domain = split_email_to_domain(data)
+        if (get_realm(domain)):
+            raise ValidationError(mark_safe("The organization with the domain already exist. Please register your account."))
```

Comment:
This should probably do something that results in the user getting a link to the registration page, right?